engines/scumm/smush/smush_player.cpp

```diff
@@ -1304,4 +1279,770 @@ void SmushPlayer::play(const char *filename, int32 speed, int32 offset, int32 st
+
+			// Check if there's the need to allocate a crossfade
+			if (_smushTracks[i].state == TRK_STATE_FADING && _smushDispatch[i].state == TRK_STATE_PLAYING) {
+				fadeStartOffset = _smushDispatch[i].audioRemaining % _smushDispatch[i].dataSize;
+				_smushDispatch[i].fadeRemaining = SMUSH_FADE_SIZE;
+				_smushDispatch[i].fadeVolume = _smushTracks[i].volume;
+				_smushDispatch[i].fadeSampleRate = _smushDispatch[i].sampleRate;
+
+				fillAudioBufWithSample(_smushTracks[i].fadeBuf, 127, _smushDispatch[i].fadeRemaining);
+				cpySize = _smushDispatch[i].dataSize - fadeStartOffset;
+
+				if (cpySize > _smushDispatch[i].fadeRemaining)
+					cpySize = _smushDispatch[i].fadeRemaining;
+				memcpy(_smushTracks[i].fadeBuf, &_smushDispatch[i].dataBuf[fadeStartOffset], cpySize);
+
+				_smushDispatch[i].volumeStep = 0;
+			} else if (_smushTracks[i].state == TRK_STATE_PLAYING) {
+				if (_smushDispatch[i].audioRemaining < _smushTracks[i].availableSize - _smushTracks[i].dataSize + 15000) {
+					if (_smushTracks[i].availableSize < _smushTracks[i].sdatSize) {
+						_smushDispatch[i].volumeStep = 0;
+						mixInFrameCount = _smushTracks[i].availableSize - _smushDispatch[i].audioRemaining - 15000;
+
+						if (mixInFrameCount > _smushDispatch[i].currentOffset)
+							mixInFrameCount = _smushDispatch[i].currentOffset;
+
+						if (_smushDispatch[i].audioRemaining + mixInFrameCount > _smushTracks[i].sdatSize - _smushDispatch[i].dataSize)
+							mixInFrameCount = _smushTracks[i].sdatSize - _smushDispatch[i].dataSize - _smushDispatch[i].audioRemaining;
+
+						if (mixInFrameCount > 0) {
+							_smushDispatch[i].fadeRemaining = SMUSH_FADE_SIZE;
+							_smushDispatch[i].fadeSampleRate = _smushDispatch[i].sampleRate;
+
+							memcpy(
+								_smushTracks[i].fadeBuf,
+								&_smushDispatch[i].dataBuf[_smushDispatch[i].audioRemaining % _smushDispatch[i].dataSize],
+								_smushDispatch[i].fadeRemaining);
+
+							_smushDispatch[i].audioRemaining += mixInFrameCount;
+							_smushDispatch[i].currentOffset -= mixInFrameCount;
+						}
+					}
+				}
+			}
+
+			// If the fade has been allocated, flush it in the mixer
+			if (_smushDispatch[i].fadeRemaining) {
+				maxFadeChunkSize = _smushDispatch[i].fadeSampleRate * feedSize / _smushAudioSampleRate;
+
+				if (_smushDispatch[i].fadeRemaining > maxFadeChunkSize) {
+					fadeRemaining = maxFadeChunkSize;
+				} else {
+					fadeRemaining = _smushDispatch[i].fadeRemaining;
+				}
+
+				fadeMixStartingPoint = 0;
+				while (fadeRemaining) {
+					fadeInFrameCount = (fadeRemaining < DIMUSE_FEEDSIZE) ? fadeRemaining : DIMUSE_FEEDSIZE;
+
+					if (fadeInFrameCount == maxFadeChunkSize) {
+						fadeFeedSize = feedSize;
+					} else {
+						fadeFeedSize = _smushAudioSampleRate * fadeInFrameCount / _smushDispatch[i].fadeSampleRate;
+					}
+
+					if (isPlayableTrack) {
+						fadeVolume = _smushDispatch[i].fadeRemaining * _smushDispatch[i].fadeVolume * _smushTrackVols[0] / 0x5F400;
```

Comment:
dividing by hex looks weird, as it's not round number :)

Reply:
Hmm, you're right. I'll see if I can find a convincing short name for turning that into a macro, like SMUSH_MAX_FADE_VOL, and convert it to base 10 anyway.